### DISCUSSION TITLE
BUG: Gitbook AI search is not indexing pages correctly

### DISCUSSION THREAD

**Discussion by ahcsco** [GitbookIO/community#601] [0. General bugs and questions]
We have set up our pilot site and noticed that the AI search is not indexing content/pages correctly and/or fully. Occasionally, the answer is "no answer could be found", when it should not be saying that. What's more strange is that the AI does not even answer the questions curated by itself as suggestions. 

Here's a demo from Gitbook's own docs site.

Content is present in Gitbook here: https://docs.gitbook.com/content-editor/blocks/cards#adding-images
A question is suggested and showing in Gitbook AI search: How can I add images to cards in GitBook?
Here's what happens:

<img width="867" alt="Screenshot 2024-04-27 at 11 15 15" src="https://github.com/GitbookIO/community/assets/112613597/46b17d5e-54eb-49be-8820-d70ade8399eb">

<img width="859" alt="Screenshot 2024-04-27 at 11 15 24" src="https://github.com/GitbookIO/community/assets/112613597/d6501528-6e7c-47fd-ab2c-5d3f917b5004">

P.S: I know the feature is in Beta, but it'd be good to fix it before general release.

**Comment by aikoii** [comment]
👋 @ahcsco! Could you please share the link to your docs or let me know your org ID?

**Reply by ahcsco** [reply]
Certainly. Our org ID is: zCWEO4HUVOjwxqT5Hla6.

**Reply by aikoii** [reply]
Thanks! We just made some changes on our end; please let me know if it is working better?

**Reply by ahcsco** [reply]
Thanks for getting back on this. So I've been trying it and I'm still seeing the same issue off and on, even with rather simple questions. Should we wait more if an indexing fix was issued or...?

https://github.com/GitbookIO/community/assets/112613597/655f4aea-5934-42a2-a8b7-01022742c153

**Reply by aikoii** [reply]
Thanks for sharing 👀! I'm looping in our team on this and will keep you posted.

**Reply by ahcsco** [reply]
To make it more interesting, we're now seeing Gitbook documentation suggestions on our site in the AI search. Just FYI, while we wait for a fix. Thank you. 
<img width="876" alt="Screenshot 2024-05-24 at 15 56 54" src="https://github.com/GitbookIO/community/assets/112613597/29b9b584-cecc-4f15-b79d-0b47be6373ba">

**Comment by ahcsco** [comment]
@aikoii I'd like to know if there has been any progress on this.

**Reply by aikoii** [reply]
Hey @ahcsco! So sorry for the delay in getting back to you.

The good news is the glitch with GitBook suggestions showing up in your docs should be fixed now!

About the AI not answering your questions – we made some changes to how things are indexed, hoping that would take care of it.  However, I just tested your docs, it looks like it didn't quite do the trick 😕. We're on it, and we'll keep digging to get this resolved for you 🙏

**Reply by ahcsco** [reply]
Thanks, @aikoii for the reply and fixes. So, I guess the "Gitbook" suggestions are not appearing in the main view (for now), but when I click a usual suggestion to have it answered, I still see some Gitbook-related suggestions in blue. Along with that yes, as you pointed out, the main problem remains that the AI is not answering suggestions. I don't know if the instance needs to be refreshed/reset to re-index all pages from scratch or what, but I guess we'll wait for a resolution from you. Thanks. 

<img width="873" alt="Screenshot 2024-07-03 at 16 56 23" src="https://github.com/GitbookIO/community/assets/112613597/9a8c9026-b588-4e8b-bf55-be05ff19cb3c">

**Reply by aikoii** [reply]
@ahcsco search should work much better now 🤞. Could you try again?

**Reply by ahcsco** [reply]
Yes definitely, it seems to be working now. Been testing for a while and have encountered a couple of "no answer found" screens from suggestions, but I guess for most part, it works as expected. This should do. 
Thanks for help. 
We can close this.

**Reply by aikoii** [reply]
That's great to hear! 🎉 Do reach out if anything comes up 🙂
